pandas/_libs/src/parser/tokenizer.c

```diff
@@ -712,15 +718,17 @@ int skip_this_line(parser_t *self, int64_t rownum) {
     }
 }
 
-int tokenize_bytes(parser_t *self, size_t line_limit, int start_lines) {
-    int i, slen;
+int tokenize_bytes(parser_t *self, size_t line_limit, int64_t start_lines) {
+    int64_t i, slen;
     int should_skip;
     char c;
     char *stream;
     char *buf = self->data + self->datapos;
 
     if (make_stream_space(self, self->datalen - self->datapos) < 0) {
-        self->error_msg = "out of memory";
+        size_t bufsize = 100;
```

Comment:
`error_msg` is _not_ pre-allocated (in the "allocated when the parser object is created sense"). It appears one is _meant_ to allocate memory for the error message just before setting it, which of course is error prone and led to the double-free issue.

Like I said, for now, I've just made sure that all uses of `error_msg` are directly preceeded by an allocation (I think two of about ten were missing them). A better approach would be to allocate a reasonable amount of space when the parser is created and not worry about allocation anywhere else, but that should be in its own PR (which I'm working on ATM).